Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/state/ThresholdedRandomCutForestState.java

```diff
@@ -22,10 +22,10 @@
 import lombok.Data;
 
 import com.amazon.randomcutforest.parkservices.state.predictorcorrector.PredictorCorrectorState;
-import com.amazon.randomcutforest.parkservices.state.preprocessor.PreprocessorState;
 import com.amazon.randomcutforest.parkservices.state.returntypes.ComputeDescriptorState;
 import com.amazon.randomcutforest.parkservices.state.threshold.BasicThresholderState;
 import com.amazon.randomcutforest.state.RandomCutForestState;
+import com.amazon.randomcutforest.state.preprocessor.PreprocessorState;
```

Comment:
Do you need to change ThresholdedRandomCutForestState.version to 4.0? I also saw other places have 3.x version.

(base) kaituo@88665a53bc93 random-cut-forest-by-aws % ag V3_
Java/core/src/main/java/com/amazon/randomcutforest/state/tree/AbstractNodeStoreMapper.java
64:        state.setVersion(Version.V3_0);

Java/core/src/main/java/com/amazon/randomcutforest/state/tree/RandomCutTreeMapper.java
67:        state.setVersion(Version.V3_0);

Java/core/src/main/java/com/amazon/randomcutforest/state/Version.java
21:    public static final String V3_0 = "3.0";
22:    public static final String V3_5 = "3.5";
23:    public static final String V3_7 = "3.7";
24:    public static final String V3_8 = "3.8";

Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java
59:        if (!state.getVersion().equals(Version.V3_0)) {
97:        state.setVersion(Version.V3_0);

Java/serialization/src/main/java/com/amazon/randomcutforest/serialize/json/v2/V2StateToV3ForestConverter.java
73:        if (!state.getVersion().equals(Version.V3_0)) {

Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/state/RCFCasterState.java
18:import static com.amazon.randomcutforest.state.Version.V3_8;
27:    private String version = V3_8;

Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/state/ThresholdedRandomCutForestState.java
18:import static com.amazon.randomcutforest.state.Version.V3_8;
35:    private String version = V3_8;

Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/state/predictorcorrector/PredictorCorrectorState.java
18:import static com.amazon.randomcutforest.state.Version.V3_8;
32:    private String version = V3_8;

Do you need to change all of them to 4.0?

Reply:
No - we do not need to change them -- we should keep them for now for both debugging and history. The logic we have followed is that when the state class is altered then we update field. Now if the state class contains a POJO which is altered then it allows us to debug where the failure is -- so in fullness of time we would probably have all of the classes changed to 4.0. But that would need care so that we do not break backward compatibility -- for example the point store mapper uses the version strongly (and changing that would need more thought -- thankfully the backward compatibility tests are now in the core as well).   I think taking these on at this point would increate more unknowns. That being said, the counterpoint -- we are making a large change, this may be the moment -- that is also true. I think if we feel strongly, we should change in subsequent PRs where we would be able to debug smaller changes!